client/senddecorator/metrics.go

```diff
@@ -0,0 +1,29 @@
+package senddecorator
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// MetricsCollector returns decorator that collects API call metrics.
+func MetricsCollector(m *prometheus.SummaryVec) autorest.SendDecorator {
+	return func(s autorest.Sender) autorest.Sender {
+		return autorest.SenderFunc(func(r *http.Request) (*http.Response, error) {
+
+			start := time.Now()
+
+			// Pass the request to next SendDecorator.
+			resp, err := s.Do(r)
+
+			elapsed := time.Since(start)
+
+			// Record latency as milliseconds.
```

Comment:
Let's use seconds as the unit please, standards recommend SI base units (see https://prometheus.io/docs/practices/naming/)

Reply:
Hmm... That's what documentation seems to recommend. Feels super counter-intuitive when expected values are in granularity of 10-100ms. I can only imagine the amount of error when doing statistics with huge number of very small floats. I guess it still works in practice?

Reply:
I think in most cases it doesn't matter, and being able to treat all 'times' as the same pays off a lot